tensorrt_llm/_torch/speculative/external_api.py

```
@@ -0,0 +1,130 @@
+                    
+                    result = await response.json()
+                    draft_tokens = self.get_nested_field_from_response(result)
+                    if len(draft_tokens) > self.max_draft_len:
+                        draft_tokens = draft_tokens[:self.max_draft_len]
+                    logger.debug(f"Retrieved draft tokens for request {request_id}")
+                    return draft_tokens
+            
+        except json.JSONDecodeError as e:
+            logger.error(f"Failed to parse JSON response for request {request_id}: {e}")
+            logger.debug(f"Raw response: {response.text[:500]}...")
+            return []
```

Ruff (0.12.2)

Line too long (127 > 120)

(E501)